addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4568,10 +4568,19 @@ var SnippetsMenu = Widget.extend({
             // TODO In future version use tool method to determine isMobilePreview.
             const mobileViewThreshold = MEDIAS_BREAKPOINTS[SIZES.LG].minWidth;
             const isMobilePreview = this.$body[0].ownerDocument.defaultView.frameElement.clientWidth < mobileViewThreshold;
+            const withOverlayEl =
+            this.options.wysiwyg.lastElement &&
+            this.options.wysiwyg.lastElement.closest(
+                ".o_snippet_mobile_invisible, .o_snippet_desktop_invisible"
+            );
             for (const invisibleOverrideEl of this.getEditableArea().find('.o_snippet_mobile_invisible, .o_snippet_desktop_invisible')) {
                 const isMobileHidden = invisibleOverrideEl.classList.contains("o_snippet_mobile_invisible");
                 invisibleOverrideEl.classList.remove('o_snippet_override_invisible');
                 if (isMobilePreview === isMobileHidden) {
+                    // Updating options upon changing preview mode to avoid ghost overlay
+                    if (invisibleOverrideEl === withOverlayEl) {
+                        this.trigger_up("snippet_option_visibility_update", {show: false});
+                    }
```

Comment:
Is it not simply:

- If the selected snippet becomes invisible after clicking the mobile/device preview button, then deactivate the snippet.

So in the end, if the selected snippet has `dataset.invisible = 1`, deactivate it.

Anyway, the code is incomplete as it is now. It works when the snippet is hidden on desktop, but not the other way -> when it is hidden on mobile.

Try this:

- Drop a snippet.
- Hide it on mobile.
- Switch to mobile preview.
- The overlay is still there (if not move your mouse everywhere), and the options are still enabled.

![image](https://github.com/user-attachments/assets/ade01a7e-7e65-48be-8304-6b77b8f72f76)